### ISSUE TITLE
Regarding NPR

### ISSUE THREAD

**Issue by Ling27** [r4dl/LAENeRF#1]
@r4dl Thanks for this great work. 
I wonder how to reproduce the non-photorealistic appearance editing results as stated in the paper? I noticed that the function train_style_step_npr is implemented however its not displayed in GUI. I would like to know how the semantic encoder influences the style transfer in NPR.  

I appreciated your reply ASAP.

**Comment by r4dl** [comment]
Hi,
thanks for your interest in our work.

Non-photorealistic appearance editing (i.e. Stylization) can be easily performed with
`bash scripts/run_{dataset}.sh scripts/configs_{dataset}/{scene}.sh -m style`

The general workflow is as demonstrated in the GUI demo video (see [here](https://www.youtube.com/watch?v=iXVr6OCD_CA&feature=youtu.be)).

Regarding the implementation: 
For details on stylization, see `nerf/utils.py/train_LAENeRF_step`:
If `style_weight > 0`:
- we first crop the region
- then, the cropped region is encoded by the style network (`editing/style_network.py/forward`)
- we then take an MSE loss with respect to the target feature (the style image as a gram matrix)

The function `train_style_step_npr` was not used to produce the results in the paper - rather, this is an experimental feature to test **LAENeRF** on stylization with a single reference image.

If you have any other issues, I am happy to help.

Best,
Lukas

**Comment by Ling27** [comment]
@r4dl Hi Hi

I have another question regarding view consistency. In the supplementary material, the view consistency metric is used to evaluate and prove that the results contain high consistency between views. 

I wonder how do you set the constraints over the model training to achieve such consistency? 

I appreciated your reply ASAP and thank you.

**Comment by r4dl** [comment]
Hi,
the view-consistency is implicitly ensured because we learn a mapping from expected ray termination to predicted color. For NeRFs, this turns out to be highly consistent across views.

In addition, due to our fine-tuning strategy, we create a multi-view consistent (stylized or recolored) training dataset - I suggest checking out the modified training dataset we generate as an intermediate output.

in short: Our stylization and recoloring framework is multi-view consistent if the base NeRF is.
